domain/controllernode/state/state_test.go

```diff
@@ -88,3 +99,85 @@ func (s *stateSuite) TestSelectDatabaseNamespace(c *gc.C) {
 	c.Check(err, jc.ErrorIs, controllernodeerrors.NotFound)
 	c.Check(namespace, gc.Equals, "")
 }
+
+func (s *stateSuite) TestSetRunningAgentBinaryVersionSuccess(c *gc.C) {
```

Comment:
This inserts right? Also need a test for update?

Reply:
The insert (curate node) is just to create a controller node such that it passes the check later that the controller node exists, in order to upsert the version in the controller_node_agent_version for that controller node.

Ref code check:
`
SELECT controller_id AS &controllerNodeAgentVersion.*
FROM controller_node
WHERE controller_id = $controllerNodeAgentVersion.controller_id
`

Reply:
I think what Ian is asking for is that you set it again after the initial set to test the upsert logic.